plugins/auth-backend/src/providers/saml/provider.ts

```diff
@@ -112,27 +113,33 @@ export class SamlAuthProvider implements AuthProviderRouteHandlers {
   }
 }
 
-type SAMLProviderOptions = {
-  entryPoint: string;
-  issuer: string;
-  path: string;
+type SAMLProviderOptions = SamlConfig & {
   tokenIssuer: TokenIssuer;
   appUrl: string;
 };
 
+type SignatureAlgorithm = 'sha1' | 'sha256' | 'sha512';
+
 export const createSamlProvider: AuthProviderFactory = ({
   providerId,
   globalConfig,
   config,
   tokenIssuer,
 }) => {
-  const url = new URL(globalConfig.baseUrl);
   const entryPoint = config.getString('entryPoint');
   const issuer = config.getString('issuer');
   const opts = {
+    callbackUrl: `${globalConfig.baseUrl}/${providerId}/handler/frame`,
     entryPoint,
     issuer,
-    path: `${url.pathname}/${providerId}/handler/frame`,
+    cert: config.getOptionalString('cert'),
+    privateCert: config.getOptionalString('privateKey'),
```

Comment:
```suggestion
    privateKey: config.getOptionalString('privateKey'),
```

Looks like `privateCert` is deprecated

Reply:
In the newer versions, yes. In 1.3.5 https://github.com/node-saml/passport-saml/tree/v1.3.5#config-parameter-details it's still privateCert.

I'll use privateCert for the code to match the API, but require the config be privateKey since we'd be upgrading the library version soon anyway.